File: web/src/api/healthChange.js
import service from '@/utils/request'

// @Tags HealthChange
// @Summary 创建HealthChange
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.HealthChange true "创建HealthChange"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /healthChange/createHealthChange [post]
export const createHealthChange = (data) => {
     return service({
         url: "/healthChange/createHealthChange",
         method: 'post',
         data
     })
 }


// @Tags HealthChange
// @Summary 删除HealthChange
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.HealthChange true "删除HealthChange"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /healthChange/deleteHealthChange [delete]
 export const deleteHealthChange = (data) => {
     return service({
         url: "/healthChange/deleteHealthChange",
         method: 'delete',
         data
     })
 }

// @Tags HealthChange
// @Summary 删除HealthChange
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除HealthChange"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /healthChange/deleteHealthChange [delete]
 export const deleteHealthChangeByIds = (data) => {
     return service({
         url: "/healthChange/deleteHealthChangeByIds",
         method: 'delete',
         data
     })
 }

// @Tags HealthChange
// @Summary 更新HealthChange
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.HealthChange true "更新HealthChange"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /healthChange/updateHealthChange [put]
 export const updateHealthChange = (data) => {
     return service({
         url: "/healthChange/updateHealthChange",
         method: 'put',
         data
     })
 }


// @Tags HealthChange
// @Summary 用id查询HealthChange
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.HealthChange true "用id查询HealthChange"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /healthChange/findHealthChange [get]
 export const findHealthChange = (params) => {
     return service({
         url: "/healthChange/findHealthChange",
         method: 'get',
         params
     })
 }


// @Tags HealthChange
// @Summary 分页获取HealthChange列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取HealthChange列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /healthChange/getHealthChangeList [get]
 export const getHealthChangeList = (params) => {
     return service({
         url: "/healthChange/getHealthChangeList",
         method: 'get',
         params
     })
 }

File: server/service/healthChange.go
package service

import (
	"gin-vue-admin/global"
	"gin-vue-admin/model"
	"gin-vue-admin/model/request"
)

// @title    CreateHealthChange
// @description   create a HealthChange
// @param     healthChange               model.HealthChange
// @auth                     （2020/04/05  20:22）
// @return    err             error

func CreateHealthChange(healthChange model.HealthChange) (err error) {
	err = global.PantaDb.Create(&healthChange).Error
	return err
}

// @title    DeleteHealthChange
// @description   delete a HealthChange
// @auth                     （2020/04/05  20:22）
// @param     healthChange               model.HealthChange
// @return                    error

func DeleteHealthChange(healthChange model.HealthChange) (err error) {
	err = global.PantaDb.Delete(healthChange).Error
	return err
}

// @title    DeleteHealthChangeByIds
// @description   delete HealthChanges
// @auth                     （2020/04/05  20:22）
// @param     healthChange               model.HealthChange
// @return                    error

func DeleteHealthChangeByIds(ids request.IdsReq) (err error) {
	err = global.PantaDb.Delete(&[]model.HealthChange{},"id in (?)",ids.Ids).Error
	return err
}

func DeleteHealthChangeByLiftIds(ids request.IdsReq) (err error) {
	err = global.PantaDb.Delete(&[]model.HealthChange{},"lift_id in (?)",ids.Ids).Error
	return err
}

// @title    UpdateHealthChange
// @description   update a HealthChange
// @param     healthChange          *model.HealthChange
// @auth                     （2020/04/05  20:22）
// @return                    error

func UpdateHealthChange(healthChange *model.HealthChange) (err error) {
	err = global.PantaDb.Save(healthChange).Error
	return err
}

// @title    GetHealthChange
// @description   get the info of a HealthChange
// @auth                     （2020/04/05  20:22）
// @param     id              uint
// @return                    error
// @return    HealthChange        HealthChange

func GetHealthChange(id uint) (err error, healthChange model.HealthChange) {
	err = global.PantaDb.Where("id = ?", id).Preload("Lift").Preload("Dimension").First(&healthChange).Error
	return
}

// @title    GetHealthChangeList
// @description   get HealthChange list by pagination, 分页获取HealthChange
// @auth                     （2020/04/05  20:22）
// @param     info            PageInfo
// @return                    error

func GetHealthChangeList(info request.HealthChangeSearch) (err error, list interface{}, total int64) {
	limit := info.PageSize
	offset := info.PageSize * (info.Page - 1)
    // 创建db
	db := global.PantaDb.Model(&model.HealthChange{})
    var healthChanges []model.HealthChange
    // 如果有条件搜索 下方会自动创建搜索语句
	err = db.Count(&total).Error
	err = db.Limit(limit).Offset(offset).Preload("Lift").Preload("Dimension").Find(&healthChanges).Error
	return err, healthChanges, total
}